src/wp-includes/revision.php

```diff
@@ -516,6 +516,53 @@ function wp_get_post_revisions( $post_id = 0, $args = null ) {
 	return $revisions;
 }
 
+/**
+ * Returns last revisions of specified post and count of total revisions.
+ *
+ * @since 6.1.0
+ *
+ * @param int|WP_Post $post Optional. Post ID or post object. Default is the global $post.
+ * @return array {
+ *     Returns associative array with last revision and total count.
+ *
+ *     @type int $revision The last revision post id or 0 if non existing.
+ *     @type int $count The total count of revisions for $post_id.
+ * }
+ */
+function wp_get_lastest_revision_id_and_total_count( $post = 0 ) {
+	$post     = get_post( $post );
+	$revision = 0;
+	$count    = 0;
+
+	if ( ! $post || ! wp_revisions_enabled( $post ) ) {
+		return compact( 'revision', 'count' );
+	}
+
+	$args = array(
+		'post_parent'            => $post->ID,
+		'fields'                 => 'ids',
+		'post_type'              => 'revision',
+		'post_status'            => 'inherit',
+		'order'                  => 'DESC',
+		'orderby'                => 'date ID',
+		'posts_per_page'         => 1,
+		'update_post_meta_cache' => false,
+		'update_post_term_cache' => false,
+		'suppress_filters'       => true,
```

Comment:
@peterwilsoncc Is there a need for `suppress_filters` filters here? I would like to avoid that where possible.

Reply:
> Is there a need for `suppress_filters` filters here?

It's for preserving backward compatibility.

`wp_get_post_revisions()` suppresses filters via the `get_posts()` defaults, so doing so here ensure that the same result is returned for this function and the following code:

```
$latest_revision_and_count = [
'count' => count( wp_get_post_revisions() ),
'revision' => wp_get_post_revisions()[0],
];
```

Reply:
There are a number of places that do not support filters for revisions. Including the revisions REST API controller. I don't see the harm here in unsuppressing the filters here.

Reply:
> There are a number of places that do not support filters for revisions.

Do you mean _suppress_?

If that's the case, then yes, let's unsuppress here too. I think that will help caching plugins.

Reply:
Agreed, I think we can safely drop this if we already have in the Revisions controller.

A passing thought, it'd be nice if we had something documented in the handbook as to when Core should be using `get_posts/suppress_filters`. Is it preferred for all new code to not suppress filters?

Reply:
> Is it preferred for all new code to not suppress filters?

That is what I have been working towards. suppress filters is about backwards compat, not an issue with new functions.